src/Workspaces/CoreTest/SolutionTests/SolutionTests.cs

```diff
@@ -658,7 +660,19 @@ private Solution CreateNotKeptAliveSolution()
             return workspace.CurrentSolution;
         }
 
-        private void StopObservingAndWaitForReferenceToGo(ObjectReference observed, int delay = 0)
+        [DllImport("dbghelp.dll")]
+        private static extern bool MiniDumpWriteDump(IntPtr hProcess, int ProcessId, IntPtr hFile, int DumpType, IntPtr ExceptionParam, IntPtr UserStreamParam, IntPtr CallbackParam);
+
+        private static void DumpProcess(string dumpFileName)
+        {
+            using (var fs = File.Create(dumpFileName))
+            {
+                var proc = System.Diagnostics.Process.GetCurrentProcess();
+                MiniDumpWriteDump(proc.Handle, proc.Id, fs.SafeFileHandle.DangerousGetHandle(), /*normal dump*/0, IntPtr.Zero, IntPtr.Zero, IntPtr.Zero);
+            }
```

Comment:
You want to get a heap dump (MiniDumpWithFullMemory?), this will only include thread's stacks.